doc/source/data/batch_inference.rst

```diff
@@ -1,823 +1,455 @@
-            :language: python
-            :start-after: __hf_no_ray_start__
-            :end-before: __hf_no_ray_end__
-
-    .. group-tab:: PyTorch
-
-        .. literalinclude:: ./doc_code/pytorch_quick_start.py
-            :language: python
-            :start-after: __pt_no_ray_start__
-            :end-before: __pt_no_ray_end__
-
-    .. group-tab:: TensorFlow
-
-       .. literalinclude:: ./doc_code/tf_quick_start.py
-           :language: python
-           :start-after: __tf_no_ray_start__
-           :end-before: __tf_no_ray_end__
-
-.. note::
-
-    As a Python user, this should all look familiar to you.
-    The only part that you might be wondering about is that we're using
-    ``Dict[str, np.ndarray]`` as input type to our ``transform`` functions.
-    We do this to ease the transition to Ray, given that Ray Data uses
-    ``Dict[str, np.ndarray]`` as the default format for its batches.
-
-
-If you can follow the above examples conceptually, you should have no trouble scaling your batch
-inference workload to a compute cluster with Ray Data.
-If you're using Ray, the three steps for running batch inference read as follows:
-
-1. Load a Ray Data dataset and apply any preprocessing you need. This will distribute your data
-   across the cluster.
-2. Define your model in a class and define a transformation that applies your model to
-   your data batches (of format ``Dict[str, np.ndarray]`` by default).
-3. Run inference on your data by using the :meth:`ds.map_batches() <ray.data.Dataset.map_batches>`
-   method from Ray Data. In this step you also define how your batch processing job
-   gets distributed across your cluster.
-
-.. note::
-
-    All advanced use cases ultimately boil down to extensions of the above three steps,
-    like loading and storing data from cloud storage, using complex preprocessing functions,
-    demanding model setups, additional postprocessing, or other customizations.
-    We'll cover these advanced use cases in the next sections.
+.. _batch_inference_quickstart:
 
-Let's scale out the above examples to a Ray cluster.
+Quickstart
+----------
 To start, install Ray with the data processing library, Ray Data:
 
 .. code-block:: bash
 
-    pip install ray[data]
-
+    pip install -U "ray[data]"
 
-1. Loading and preprocessing data
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Using Ray Data for offline inference involves four basic steps:
 
-For this quick start guide we use very small, in-memory datasets by
-leveraging common Python libraries like NumPy and Pandas.
-
-In fact, we're using the exact same datasets as in the previous section, but load
-them into Ray data.
-The result of this step is a Dataset ``ds`` that we can use to run inference on.
+| **Step 1:** Load your data into a Ray Dataset. Ray Data supports many different data types and formats. See :ref:`Loading Data <loading_data>` for more details.
+| **Step 2:** Load the pre-trained model in a Python class. 
```

Comment:
This sounds like we are already "loading" the model into memory in this step. Something like "Define a Python class that will load a pre-trained model and use it for inference" would be more clear.